### PR TITLE
feat: ship command

### PR DESCRIPTION
**Pull request by favna**
Ship command, ported from Ribbon. but better.

### CODE REVIEW

File: src/commands/Misc/ship.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			description: language => language.tget('COMMAND_SHIP_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_SHIP_EXTENDED'),
+			requiredPermissions: ['ATTACH_FILES'],
+			runIn: ['text'],
+			usage: '(firstUser:firstUser) (secondUser:secondUser)',
+			usageDelim: ' '
+		});
+
+		this.createCustomResolver('firstUser', (arg, possible, message) => {
+			if (!arg) return message.guild?.members.random()!.user;
```
Comment (vladfrangu) [inline src/commands/Misc/ship.ts]: Won't this `?.` generate code that is..useless? We'll always have message.guild in this context

File: src/commands/Misc/ship.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			return this.userNameArgument.run(arg, possible, message);
+		});
+
+		this.createCustomResolver('secondUser', (arg, possible, message) => {
+			if (!arg) return message.guild?.members.random()!.user;
```
Comment (vladfrangu) [inline src/commands/Misc/ship.ts]: Ditto

File: src/commands/Misc/ship.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		const darkTheme = message.author.settings.get(UserSettings.DarkTheme);
+		const shipName = removeConfusables(
+			firstUser.username.substring(0, roundNumber(firstUser.username.length / roundNumber((Math.random() * 4) + 2))) +
+			secondUser.username.substring(roundNumber(secondUser.username.length / roundNumber((Math.random() * 4) + 2)))
+		).replace(this.removeSymbolsRegex, '');
+
+		const DATA = message.language.tget('COMMAND_SHIP_DATA');
+		const attachment = await new Canvas(224, 88)
+			.save()
+			.addImage(darkTheme ? this.darkThemeTemplate! : this.lightThemeTemplate!, 0, 0, 224, 88)
```
Comment (vladfrangu) [inline src/commands/Misc/ship.ts:71]: You could simplify this by removing the `!` from the end, and where you define the buffers:
```ts
private darkThemeTemplate!: Buffer;
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — DISMISSED** [review]

File: src/commands/Misc/ship.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+import { fetch, FetchResultTypes, getDisplayAvatar, loadImage } from '@utils/util';
+import { Image } from 'canvas';
+import { Canvas } from 'canvas-constructor';
+import { remove as removeConfusables } from 'confusables';
+import { CommandStore, KlasaMessage } from 'klasa';
+import { join } from 'path';
+
+export default class extends SkyraCommand {
+
+	private removeSymbolsRegex = /(?:[~`!@#%^&*(){}[\];:"'<,.>?/\\|_+=-])+/g;
```
Comment (kyranet) [inline src/commands/Misc/ship.ts]:
```suggestion
	private readonly kRemoveSymbolsRegex = /(?:[~`!@#%^&*(){}[\];:"'<,.>?/\\|_+=-])+/g;
```

And update all references

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**